snap/snapshots.go

```diff
@@ -0,0 +1,110 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snap
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+var osOpen = os.Open
+
+// SnapshotOptions contains the information provided by a snap package about
+// what files should be included in the snapshots.
+type SnapshotOptions struct {
+	// ExcludePaths is the list of file and directory patterns that need to be
+	// excluded from a snapshot. At the moment the only supported globbing
+	// character is "*", which stands for any sequence of characters other than
+	// "/".
+	ExcludePaths []string `yaml:"exclude"`
+}
+
+// ReadSnapshotYaml reads the snapshot manifest file for the given snap.
+func ReadSnapshotYaml(si *Info) (*SnapshotOptions, error) {
+	file, err := osOpen(filepath.Join(si.MountDir(), "meta", "snapshots.yaml"))
+	if os.IsNotExist(err) {
+		return &SnapshotOptions{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	return readSnapshotYaml(file)
+}
+
+// ReadSnapshotYaml reads the snapshot manifest file for the given snap
+// container.
+func ReadSnapshotYamlFromSnapFile(snapf Container) (*SnapshotOptions, error) {
+	sy, err := snapf.ReadFile("meta/snapshot.yaml")
```

Comment:
oops, typo in the filename, the tests have the typo too.  maybe this should a const, anyway I think we want snapshots.yaml